scripts/gen-identity.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"crypto/rand"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/peer"
+)
+
+const (
+	encFileName = "identity.key.encrypted"
+	sopsConfig  = ".sops.yaml"
+)
+
+func main() {
+	flag.Usage = usage
+	flag.Parse()
+
+	err := genIdentity()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+func usage() {
+	name := filepath.Base(os.Args[0])
+	fmt.Fprintln(os.Stderr, "NAME:")
+	fmt.Fprintln(os.Stderr, "   ", name)
+	fmt.Fprintln(os.Stderr)
+	fmt.Fprintln(os.Stderr, "USAGE:")
+	fmt.Fprintln(os.Stderr, "    go run", name, "[command options]")
+	fmt.Fprintln(os.Stderr)
+	fmt.Fprintln(os.Stderr, "DESCRIPTION:")
+	fmt.Fprintln(os.Stderr, "    Generate an encrypted identity file, identity.key.encrypted, and output the associated peer ID.")
+	fmt.Fprintln(os.Stderr, "    Run", name, "from a directory with a", sopsConfig, "file to generate the encrypted identity.")
+	fmt.Fprintln(os.Stderr)
+	fmt.Fprintln(os.Stderr, "    Example:")
+	fmt.Fprintln(os.Stderr, "        cd deploy/manifests/dev/us-east-2/tenant/storetheindex")
+	fmt.Fprintf(os.Stderr, "        go run ../../../../../../scripts/%s\n", name)
+	fmt.Fprintln(os.Stderr, "        mv", encFileName, "instances/my-indexer/")
+	fmt.Fprintln(os.Stderr)
+	fmt.Fprintln(os.Stderr, "OPTIONS:")
+	fmt.Fprintln(os.Stderr, "    -help, -h       Show help")
+}
+
+func genIdentity() error {
+	if !fileExists(sopsConfig) {
+		return fmt.Errorf("must be in directory with sops config %q", sopsConfig)
+	}
+
+	pkFileName, pid, err := genPrivateKeyFile()
+	if err != nil {
+		return err
+	}
+	defer os.Remove(pkFileName)
+
+	// sops -e identity.key > identity.key.encrypted
+	cmd := exec.Command("sops", "-e", pkFileName)
```

Comment:
I advise against this TBH. It is easy enough to pipe it to sops and having it here leaks infra choices. So my vote is for something like [this](https://github.com/ipni/storetheindex/pull/341).